app/commands/decidim/anonymous_codes/admin/create_tokens.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Decidim
+  module AnonymousCodes
+    module Admin
+      class CreateTokens < Decidim::Command
+        def initialize(form, code_group)
+          @form = form
+          @code_group = code_group
+        end
+
+        def call
+          return broadcast(:invalid) if form.invalid?
+
+          transaction do
+            create_code_group!
+          end
```

Comment:
As this SQL is already atomic, there's no need to specify a transaction here, so you can simply
```suggestion
          Token.create!(group: code_group, token: form.token)
```

Reply:
Thanks